### PR TITLE
optimize DivRem

### PR DESCRIPTION
**Pull request by Hecate2**
Using MOD instead of l-l/r*r
Original implementation check and throw when the remainder overflows. However it is impossible for the remainder to overflow, because
```python
if MinValue < 0
  abs(remainder) < abs(MinValue)
if MinValue == 0
  0 < remainder < MaxValue
```
Therefore I cancelled the check.
But it is possible for the quotient to overflow in a single case, when the allowed range is [-128, 127] and I call DivRem(-128, -1). I did not check and throw in this case, because it is rare and expensive, and the original implementation did not check it.

### CODE REVIEW

**Comment by Jim8y** [comment]
@Hecate2 please fix conflict

File: src/Neo.Compiler.CSharp/MethodConvert/System/SystemCall.BigInteger.cs [hunk 1]
```diff
@@ -488,21 +488,22 @@ private static void HandleMathBigIntegerDivRem(MethodConvert methodConvert, Sema
...
         // Perform division
-        methodConvert.AddInstruction(OpCode.DUP);
-        methodConvert.AddInstruction(OpCode.ROT);
-        methodConvert.AddInstruction(OpCode.TUCK);
-        methodConvert.AddInstruction(OpCode.DIV);
+        methodConvert.AddInstruction(OpCode.DUP); // r, l, l
+        methodConvert.Push(2);
+        methodConvert.AddInstruction(OpCode.PICK);// r, l, l, r
+        methodConvert.AddInstruction(OpCode.DIV);  // r, l, l/r
+        // TODO: for types that is restricted by range, check l/r <= MaxValue
```
Comment (Jim8y) [inline src/Neo.Compiler.CSharp/MethodConvert/System/SystemCall.BigInteger.cs:497]: Are you going to work on this todo？

Comment (Hecate2) [reply src/Neo.Compiler.CSharp/MethodConvert/System/SystemCall.BigInteger.cs:497]: No. I think we can just allow overflow.

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/MethodConvert/System/SystemCall.BigInteger.cs:497]: Will this be the same as C# behavior?

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/MethodConvert/System/SystemCall.BigInteger.cs:497]: then can you please explain this as comment where others can learn.

Comment (Hecate2) [reply src/Neo.Compiler.CSharp/MethodConvert/System/SystemCall.BigInteger.cs]: > then can you please explain this as comment where others can learn.

Done

**Review by shargon — APPROVED** [review]

**Review by Jim8y — APPROVED** [review]

**Pull request merged**